docs/implementation-guides/building-integrations/test-functions.mdx

```diff
@@ -0,0 +1,224 @@
+---
+title: 'Test functions'
+sidebarTitle: 'Test functions'
+description: 'How to test your syncs and actions locally'
+---
+
+Testing your integration functions ensures they work correctly before deploying to production. Nango provides two testing modes:
+
+- **Remote testing** (default): Tests with real API calls and validates output against your schema
+- **Local testing**: Tests with mocked API responses for fast, repeatable regression testing
+
+## Remote testing
+
+Remote testing executes your function making real API calls to verify it works end-to-end.
+
+### Run a remote test
+
+```bash
+nango test <script-name> <connection-id>
+```
+
+This command:
+1. Compiles your function
+2. Executes it against a real connection
+3. Validates output against your model schema
+4. Reports any validation errors
+
+Example:
+```bash
+nango test salesforce-contacts conn_123abc
+```
+
+<Tip>
+By default, tests use connections from your `dev` environment. Change this with the `-e` flag.
+</Tip>
+
+### Pass input to actions
+
+For actions that require input, use the `--input` flag:
+
+```bash
+nango test create-contact conn_123abc --input '{"email": "test@example.com"}'
+```
+
+You can also load input from a file:
+```bash
+nango test create-contact conn_123abc --input @fixtures/contact.json
+```
+
+## Local testing
+
+Local testing uses mocked API responses to test your functions without making real API calls. This is useful for:
+
+- Fast iteration during development
+- Regression testing in CI/CD
+- Testing without rate limits
+- Reproducible test results
+
+### Set up local testing
+
+**Step 1: Capture API responses**
+
+First, save real API responses from a working connection:
+
+```bash
+nango dryrun <script-name> <connection-id> --save-responses
```

Comment:
should we have an alias `nango test <script-name> <connection-id> --record` to avoid having to juggle between different subcommand